specs/duvet-specification.md

```diff
@@ -0,0 +1,475 @@
+and this punctuation MUST NOT terminate the requirement by
+excluding the following elements from the list of requirements.
+
+In the case of the requirement being terminated by a table,
+the text proceeding the table SHOULD be concatenated
+with each row of the table to form a requirement.
+Table cells MAY have RFC 2119 keywords,
+this is the same as regular sentences with multiple keywords.
+Table cells MAY contain a period (.) or exclamation point (!)
+and this punctuation MUST NOT terminate the requirement
+by excluding the following rows from the table of requirements.
+
+### Legacy Requirement
+
+Older versions of Duvet were more restrictive in parsing requirements.
+They did not treat elements of lists or rows in a table as individual elements.
+For backwards compatibility Duvet MUST support
+this older simpler form of requirement identification.
+Any complete sentence containing at least one RFC 2119 keyword MUST be treated as a requirement.
+A requirement MAY contain multiple RFC 2119 keywords.
+A requirement MUST be terminated by one of the following:
+
+- period (.)
+- exclamation point (!)
+- an empty blank line
+
+The main distinction between this legacy and regular requirement identification
+is that there is no sugar for lists or tables.
+For a given a specification Duvet MUST use one way to identify requirements.
+
+### Formats
+
+Duvet MUST be able to parse specifications formatted as:
+
+- Markdown
+- IETF style RFCs as text files.
+
+#### Requirements to TOML
+
+Duvet SHOULD be able to record parsed requirements into Toml Files.
+
+These Toml features supports users of Duvet who do not author the specifications they are implementing.
+As such, they need to customize the extracted requirements,
+modifying the content of requirements or adding/removing requirements.
+
+## Annotation
+
+Annotations are references to a text from a section in a specification,
+written as comment in the source code and test code.
+Annotations are generally stored as formatted comments in source within a project.
+
+### Meta
+
+The default identifier for the meta part in source documents MUST be //= followed by a single space.
+This identifier of meta parts MUST be configurable.
+
+### Meta: Location
+
+The first line of the meta part identifies the location of the content,
+it MUST be parsed as a URL.
+All parts of the URL other than a URL fragment MUST be optional
+and MUST identify the specification that contains this section and content.
+The URL MUST contain a URL fragment that uniquely identifies
+the section that contains this content.
+If the URL only contains a URL fragment
+then this content only exists as an annotation.
+Such comments are useful to use Duvet to manage implementation specific requirements
+that do not exist in a specification.
+
+The Meta: Locations for Annotations targeting specifications
+written in Markdown will NOT be identical to Locations targeting specifications written in IETF.
+
```

Comment:
cc @texastony @camshaft I think that this resolves the conversation about order.
Feel free to merge

```suggestion

### Meta: Key/Values parsing

After the [Meta: Location](#meta-location) all additional meta data is a series of name/value pairs.
The name MUST be the characters between the meta identifier and the first `=`.
The value MUST be all characters after the first `=`.
If consecutive duplicate names exist in a meta section
the values MUST be concatenated with a `\n`.
```